logbook-core/src/main/java/org/zalando/logbook/HeaderFilters.java

```diff
@@ -30,6 +30,27 @@ public static HeaderFilter replaceHeaders(final BiPredicate<String, String> pred
         return eachHeader((key, value) -> predicate.test(key, value) ? replacement : value);
     }
 
+    public static HeaderFilter removeHeaders(final Predicate<String> keyPredicate) {
+        return removeHeaders((key, value) -> keyPredicate.test(key));
+    }
+
+    public static HeaderFilter removeHeaders(final BiPredicate<String, String> predicate) {
+        return headers -> {
+            final BaseHttpMessage.HeadersBuilder result = new BaseHttpMessage.HeadersBuilder();
+
+            for (final Map.Entry<String, List<String>> e : headers.entrySet()) {
+                for (final String value : e.getValue()) {
+                    final String key = e.getKey();
+                    if (!predicate.test(key, value)) {
+                        result.put(key, value);
```

Comment:
I would also change `eachHeader` to use `.forEach(v -> result.put(k, v))` and maybe name local variables here and there in the same style